File: boj/#1926.py
from collections import deque

way = [[0, 1], [1, 0], [-1, 0], [0, -1]]


def BFS(i, j):
    size = 1

    q = deque()
    board[i][j] = 0
    q.append([i, j])

    while len(q) > 0:
        curi, curj = q.popleft()

        for di, dj in way:
            nexti, nextj = curi + di, curj + dj

            if nexti < 0 or nextj < 0 or nexti >= N or nextj >= M:
                continue

            if board[nexti][nextj] == 1:
                size += 1
                board[nexti][nextj] = 0
                q.append([nexti, nextj])

    return size



N, M = list(map(int, input().split()))
board = [list(map(int, input().split())) for _ in range(N)]

pic_cnt = 0
pic_max = 0

for i in range(0, N):
    for j in range(0, M):
        if board[i][j] == 1:
            pic_cnt += 1
            pic_max = max(pic_max, BFS(i, j))

print(pic_cnt)
print(pic_max)

File: boj/#1926.cpp
#include<iostream>
#include<queue>
#include<algorithm>

using namespace std;

#define MAX 501
#define DIR 4


int N, M;
int dir[4][2] = { {1, 0}, {0, 1}, {-1, 0}, {0, -1} };
bool board[MAX][MAX];

// 2264KB, 32 ms
int BFS(int i, int j) {
	
	int size = 1;
	queue<pair<int, int>> q;
	
	board[i][j] = 0;
	q.push({ i, j });

	while (!q.empty()) {
		int curi = q.front().first;
		int curj = q.front().second;
		q.pop();

		for (int d = 0; d < DIR; d++) {
			int nexti = curi + dir[d][0];
			int nextj = curj + dir[d][1];

			if (nexti < 0 || nextj < 0 || nexti >= N || nextj >= M)
				continue;

			if (board[nexti][nextj] == 1) {
				size++;
				board[nexti][nextj] = 0;
				q.push({ nexti, nextj });
			}
		}
	}

	return size;
}


int tmp_size = 0;

// 13865 KB, 32 ms
void DFS(int i, int j) {
	board[i][j] = 0;
	tmp_size++;
	for (int d = 0; d < DIR; d++) {
		int nexti = i + dir[d][0];
		int nextj = j + dir[d][1];

		if (nexti < 0 || nextj < 0 || nexti >= N || nextj >= M)
			continue;

		if (board[nexti][nextj] == 1)
			DFS(nexti, nextj);
	}
}

int main() {
	
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	cin >> N >> M;

	for (int i = 0; i < N; i++) {
		for (int j = 0; j < M; j++) {
			cin >> board[i][j];
		}
	}

	int pic_cnt = 0;
	int pic_max = 0;

	/*for (int i = 0; i < N; i++) {
		for (int j = 0; j < M; j++) {
			if (board[i][j] == 1) {
				pic_cnt++;
				pic_max = max(pic_max, BFS(i, j));
			}
		}
	}*/

	for (int i = 0; i < N; i++) {
		for (int j = 0; j < M; j++) {
			if (board[i][j] == 1) {
				pic_cnt++;
				tmp_size = 0;
				DFS(i, j);
				pic_max = max(pic_max, tmp_size);
			}
		}
	}

	cout << pic_cnt << "\n" << pic_max;

	return 0;
}